src/java.security.jgss/share/classes/sun/security/krb5/internal/crypto/CksumType.java

```diff
@@ -43,82 +43,71 @@ public static CksumType getInstance(int cksumTypeConst)
-            break;
-
-        case Checksum.CKSUMTYPE_HMAC_SHA1_96_AES256:
-            cksumType = new HmacSha1Aes256CksumType();
-            cksumTypeName =
-                "sun.security.krb5.internal.crypto.HmacSha1Aes256CksumType";
-            break;
-
-        case Checksum.CKSUMTYPE_HMAC_SHA256_128_AES128:
-            cksumType = new HmacSha2Aes128CksumType();
-            cksumTypeName =
-                    "sun.security.krb5.internal.crypto.HmacSha2Aes128CksumType";
-            break;
-
-        case Checksum.CKSUMTYPE_HMAC_SHA384_192_AES256:
-            cksumType = new HmacSha2Aes256CksumType();
-            cksumTypeName =
-                    "sun.security.krb5.internal.crypto.HmacSha2Aes256CksumType";
-            break;
-
-        case Checksum.CKSUMTYPE_HMAC_MD5_ARCFOUR:
-            cksumType = new HmacMd5ArcFourCksumType();
-            cksumTypeName =
-                "sun.security.krb5.internal.crypto.HmacMd5ArcFourCksumType";
-            break;
-
-            // currently we don't support MD4.
-        case Checksum.CKSUMTYPE_RSA_MD4_DES_K:
-            // cksumType = new RsaMd4DesKCksumType();
-            // cksumTypeName =
-            //          "sun.security.krb5.internal.crypto.RsaMd4DesKCksumType";
-        case Checksum.CKSUMTYPE_RSA_MD4:
-            // cksumType = new RsaMd4CksumType();
-            // linux box support rsamd4, how to solve conflict?
-            // cksumTypeName =
-            //          "sun.security.krb5.internal.crypto.RsaMd4CksumType";
-        case Checksum.CKSUMTYPE_RSA_MD4_DES:
-            // cksumType = new RsaMd4DesCksumType();
-            // cksumTypeName =
-            //          "sun.security.krb5.internal.crypto.RsaMd4DesCksumType";
-
-        default:
+            case Checksum.CKSUMTYPE_HMAC_SHA1_96_AES128:
+                cksumType = new HmacSha1Aes128CksumType();
+                cksumTypeName =
+                        "sun.security.krb5.internal.crypto.HmacSha1Aes128CksumType";
+                break;
+
+            case Checksum.CKSUMTYPE_HMAC_SHA1_96_AES256:
+                cksumType = new HmacSha1Aes256CksumType();
+                cksumTypeName =
+                        "sun.security.krb5.internal.crypto.HmacSha1Aes256CksumType";
+                break;
+
+            case Checksum.CKSUMTYPE_HMAC_SHA256_128_AES128:
+                cksumType = new HmacSha2Aes128CksumType();
+                cksumTypeName =
+                        "sun.security.krb5.internal.crypto.HmacSha2Aes128CksumType";
+                break;
+
+            case Checksum.CKSUMTYPE_HMAC_SHA384_192_AES256:
+                cksumType = new HmacSha2Aes256CksumType();
+                cksumTypeName =
+                        "sun.security.krb5.internal.crypto.HmacSha2Aes256CksumType";
+                break;
+        }
+        if (cksumType == null) {
             throw new KdcErrException(Krb5.KDC_ERR_SUMTYPE_NOSUPP);
```

Comment:
Could we add the checksum type that is disabled/not supported to this exception message?